### PR TITLE
fix: TT-226 Show user an error message when they try to save a time e…

### PR DESCRIPTION
**Pull request by eduardisrael**
[TT-226](https://ioetec.atlassian.net/browse/TT-226)

This PR implement 'When the user tries to save, an error message should be shown that the clock out time can’t be before the clock in time'.

### CODE REVIEW

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts [hunk 1]
```diff
@@ -576,6 +576,12 @@ describe('DetailsFieldsComponent', () => {
     const foundActivity = component.findInactiveActivity(state.activities.data);
     expect(foundActivity).toEqual(expectedActivity);
   });
+
+  it('should return false when the start time entry is not greater than the end time', () => {
+    const result = component.isStartTimeEntryAfterEndedEntry();
+
+    expect(result).toBeFalse();
+  });
```
Comment (thegreatyamori) [inline src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:584]: In this part I think it we need more test cases to ensure that the output will not fail. I've prepared a table with the all combinations and outputs.

I would suggest to select the most relevant cases. 😄 

image

**Review by thegreatyamori — COMMENTED** [review]
Good Job !!

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts (hunk 1 above)
Comment (LEON12699) [reply src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:584]: For what @thegreatyamori  recommends you might consider using ` parameterized tests` similar to those found in line 538.
You could pass the values to the form using the patchValue something like this:

```
const FormDateValues = {
      start_date: '2021-05-15',
      end_date: '2021-04-14',
      start_hour: '15:05',
      end_hour: '17:00',
  };

    component.entryForm.patchValue(FormDateValues);
    const result = component.isStartTimeEntryAfterEndedEntry();
```

**Comment by LEON12699** [comment]
LGTM 🚀 !

**Review by thegreatyamori — APPROVED** [review]

**Review by LEON12699 — APPROVED** [review]

**Pull request merged**